content/v2.1/helpers/form_helper.md

```diff
@@ -0,0 +1,964 @@
+
+Please read this section for the methods for building the form's fields.
+
+## form_for
+
+Builds the form using the given base name for all fields.
+The overall structure of the attributes available is as follows:
+
+### Usage
+
+```ruby
+form_for(base_name, url, values: _form_for_values, params: _form_for_params, **attributes)
+```
+
+**Basic Example**
+
+```ruby
+<%= form_for("book", "/books", class: "form-horizontal") do |f| %>
+  <div>
+    <%= f.label "title" %>
+    <%= f.text_field "title", class: "form-control" %>
+  </div>
+
+  <%= f.submit "Create" %>
+<% end %>
+```
+
+This will render HTML as shown below:
+
+```html
+<form action="/books" method="POST" accept-charset="utf-8" class="form-horizontal">
+  <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+  <div>
+    <label for="book-title">Title</label>
+    <input type="text" name="book[title]" id="book-title" value="Test Driven Development">
+  </div>
+
+  <button type="submit">Create</button>
+</form>
+```
+
+As you can see, all the fields had been prefixed with the form's name, which is a string `book`, passed as a first argument.
+
+To show CSRF meta tags, you need to have [sessions enabled](/v2.0/actions/sessions). Without that, you won't have the `csrf_token` hidden input field available.
+
+**Skipping the form name**
+
+If you don't want to use the form name, skip it, and pass a URL as a first argument.
+
+```ruby
+<%= form_for("/books", class: "form-horizontal") do |f| %>
+  <div>
+    <%= f.label "books.title" %>
+    <%= f.text_field "books.title", class: "form-control" %>
+  </div>
+
+  <%= f.submit "Create" %>
+<% end %>
+```
+
+This way you get a full control over naming each of the form's element
+
+```html
+<form action="/books" method="POST" accept-charset="utf-8" class="form-horizontal">
+  <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+  <div>
+    <label for="book-title">Title</label>
+    <input type="text" name="book[title]" id="book-title" value="Test Driven Development">
+  </div>
+
+  <button type="submit">Create</button>
+</form>
+```
+
+**Method overrides**
+
+While the default HTTP request method is set to `POST`, you can override it by passing in the `method:` keyword.
+
+```ruby
+<%= form_for("/books/123", method: :put) do |f| %>
+ <%= f.text_field "book.title" %>
+ <%= f.submit "Update" %>
+<% end %>
+```
+
+This will override the method giving you the control of the type of action you want to perform.
+
+```ruby
+<form action="/books/123" accept-charset="utf-8" method="POST">
+ <input type="hidden" name="_method" value="PUT">
+ <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+ <input type="text" name="book[title]" id="book-title" value="Test Driven Development">
+
+ <button type="submit">Update</button>
+</form>
+```
+
+**Overriding form values**
```

Comment:
@swilgosz In the absence of error handling examples, I'm for skipping this section.
This gives the reader the false impression that she/he needs to manually build that hash in case of failed new/edit forms.

Reply:
I'm not sure about it. I see playing with form values to pre-fill input fields is a useful knowledge. Maybe we could clarify/reword this section instead?